Non existent test resource directory fails `nativeTest`

### Expected Behavior

Having no `src/test/resources` directory should not fail the build

### Actual Behaviour

```
> Task :generateTestResourcesConfigFile FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateTestResourcesConfigFile'.
> java.nio.file.NoSuchFileException: /Users/graemerocher/dev/micronaut/apps/quick-test/build/resources/test
```

### Steps To Reproduce

Create a new micronaut project with gradle plugin 3.0.0-M1 and run `gradle nativeTest`

### Environment Information

- Gradle Plugin 3.0.0-M1

### Example Application

_No response_

### Version

Any

--- a/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
@@ -20,8 +20,10 @@ import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.SourceSetContainer;
 import org.gradle.api.tasks.TaskContainer;
 import org.gradle.api.tasks.testing.Test;
+import org.gradle.language.jvm.tasks.ProcessResources;
 import org.gradle.process.CommandLineArgumentProvider;
 
+import java.io.File;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashSet;
@@ -43,6 +45,7 @@ public class MicronautGraalPlugin implements Plugin<Project> {
     @Override
     public void apply(Project project) {
         project.getPluginManager().apply(NativeImagePlugin.class);
+        workaroundForResourcesDirectoryMissing(project);
         project.getPluginManager().withPlugin("io.micronaut.library", plugin -> {
             MicronautExtension extension = project.getExtensions().findByType(MicronautExtension.class);
             configureMicronautLibrary(project, extension);
@@ -74,6 +77,25 @@ public class MicronautGraalPlugin implements Plugin<Project> {
                 registerTestAgainstNativeImageTask(alreadyRegisteredTaskNames, tasks, testName);
             }));
         });
+    }
+
+    private void workaroundForResourcesDirectoryMissing(Project project) {
+        project.getPluginManager().withPlugin("java", plugin ->
+                project.afterEvaluate(unused -> {
+                    // Workaround for https://github.com/graalvm/native-build-tools/issues/175
+                    // and https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/306
+                    project.getTasks().withType(ProcessResources.class, task -> {
+                        // yes we do this at config time, because otherwise the workaround
+                        // simply doesn't work because there would be no inputs so that
+                        // task would never be executed. So yes, this is incorrect because
+                        // the destination dir _could_ be changed after this is executed
+                        File destinationDir = task.getDestinationDir();
+                        if (destinationDir != null) {
+                            destinationDir.mkdirs();
+                        }
+                    });
+                })
+        );
     }
 
     /**

